### ISSUE TITLE
Socks5 is completely butchered for user/pass auth

### ISSUE THREAD

**Issue by vans163** [cmullaparthi/ibrowse#143]
I make a fork to fix it but its completely butchered in master, I cant even follow the code, its a mess. It does not work at all.  

Currently going to fix the older release thats 1-2 years old which works to a degree just the user/pass handshake is broken.

No point in pushing that as its very diff from master.

To clean it up I can recommend QT style proxy config which is:
```
type= socks5|socks4|https, host= ""|<<>>, port= number(), user= ""|<<>>, pass= ""|<<>>
```

Part of the problem now is parsing proxy_host for https and socks5_host for socks5 is a huge mess.

**Comment by cmullaparthi** [comment]
I'm sorry, but I don't understand what you are saying. Can you please explain?

**Comment by vans163** [comment]
I dont think you have tried using master branch for a socks5 connection. It needs to be removed from documentation.  As people are googling socks5 client erlang and finding your library, for which master branch is completely broken and does not work at all with socks5.

**Comment by cmullaparthi** [comment]
I posted a working example earlier. Stop being rude and explain what is t working. 

--
W: http://chandrusoft.wordpress.com

**Comment by vans163** [comment]
I am not being rude I am being to the point and blunt. I don't have time nor desire to cherry coat.

Its like talking to a wall.  Had you actually tested a socks5 proxy you would see its not working.

I just said in two different issues what is not working:

```
#1 socks5 for HTTP post/get is not working. At all.

#2 socks5 for non-http conns is working now (with your latest commit) if you call directly into ibrowse socks5 code.

#3 socks5 for non-http conns is not parsing socks5_pass, it expects socks5_password (docs need to be adjusted)

#4 socks5 for non-http conns has a broken auth handshake when a user/password is provided.
```

**Comment by cmullaparthi** [comment]
Fixed in version 4.2.3
